src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs

```diff
@@ -122,8 +131,9 @@ internal static async Task<bool> IsOverrideAsync(Solution solution, ISymbol memb
                         // In this case, Base.Goo *does* implement IGoo.Goo in the context of the type
                         // Derived.
                         var containingType = symbol.ContainingType.OriginalDefinition;
-                        var derivedClasses = await SymbolFinder.FindDerivedClassesAsync(
-                            containingType, solution, projects, cancellationToken).ConfigureAwait(false);
+                        var derivedClasses = includeImplementationsThroughDerivedTypes
+                            ? await FindDerivedClassesAsync(containingType, solution, projects, cancellationToken).ConfigureAwait(false)
+                            : SpecializedCollections.EmptyEnumerable<INamedTypeSymbol>();
```

Comment:
this is the important bit.  we were always walking downwards looking for interface-impls caused by derived classes.  This allows us to skip this expensive part in doc-highlights.